webodf/lib/core/DomUtils.js

```diff
@@ -581,15 +581,21 @@
          * Removes all unwanted nodes from targetNode includes itself.
          * @param {!Node} targetNode
          * @param {function(!Node):!boolean} shouldRemove check whether a node should be removed or not
+         * @param {function(!Node):!boolean=} shouldSkip check whether a node and its subtree should be skipped
          * @return {?Node} parent of targetNode
          */
-        function removeUnwantedNodes(targetNode, shouldRemove) {
+        function removeUnwantedNodes(targetNode, shouldRemove, shouldSkip) {
```

Comment:
Hm, I see your point. Indeed would make this method more flexibel if the callee has control about how shouldSkip & shouldDelete relate to each other. Will give it a try.